### PR TITLE
refactor: extract pure helpers from the capture+format god-classes

### PR DESCRIPTION
**Pull request by chhoumann**
## Summary

Behavior-preserving architecture pass on the three biggest / most-central files of the capture + format pipeline. One pattern throughout — **functional core, injected effects**: pure logic moves into independently-testable helper modules; the classes stay thin stateful adapters that bind instance state / vault probes and route side-effects.

No user-facing behavior changes. No new repo docs. Commits are `refactor(...)` so semantic-release does not cut a release.

## Changes

| Commit | What | Effect |
|---|---|---|
| `refactor(capture): extract document-positioning…` | `captureChoiceFormatter.ts` → new `formatters/helpers/insertionPositioning.ts` (17 pure fns: target matching, splice, blank-line spacing, ordered splice, fence/frontmatter masking) | **1324 → 989 lines** (−25%). Splice fns return `{ content, insertedEndOffset }`; the class routes the offset through its validating `setCaptureInsertionEndOffset` so the cursor side-effect is preserved. |
| `refactor(capture): extract capture-target classifier…` | `CaptureChoiceEngine.ts` → new `engine/helpers/captureTargetResolution.ts` (pure `resolveCaptureTarget` with injected vault probes) | Removes the `(engine as any).resolveCaptureTarget` test smell; the classifier is now a first-class testable unit. UI-coupled selection stays in the engine. |
| `refactor(format): extract FILE-token value rendering…` | base `Formatter` (`formatter.ts`) → new `formatters/helpers/fileTokenRendering.ts` (`renderStoredFileValue`, link generation injected) | The vault-coupled `getFileLinkForStoredValue` stays in the class and is injected as `resolveLink`. |

## Verification

- `pnpm run build` + `pnpm run lint` green.
- `pnpm run test`: **2956 passing** (+52 new tests across the 3 helper modules; 0 regressions vs the 2904 baseline).
- **Live-tested in the real app** (isolated worktree e2e vault) after each phase and again on the final reloaded build:
  - ordered-insert capture → `## 2026-06-15` lands correctly between `16`/`14`, frontmatter intact;
  - `{{FILE:..|name/path/link}}` → `name=MyNote path=Inbox/MyNote.md link=[[MyNote]]`;
  - `dev:errors` clean throughout.
- Adversarial-reviewed (opposite-model): Phase 1 Skeptic PASS + Architect (one real finding — leftover dead methods — fixed); Phase 2 Skeptic PASS; Phase 3's reviewer hung and was terminated, so it was verified via self-review + the two identical prior passes + the test/live evidence.

## Not in scope (deferred, documented)

Higher-risk deeper surgery / polish, intentionally left out to avoid regressing a stable plugin: breaking up the 270-line `CaptureChoiceEngine.run()`; restructuring the formatter inheritance and the `RequirementCollector extends Formatter` prompt-interception coupling; unifying `SpliceResult` with `noteContentInsertion`'s `InsertionResult`; a shared markdown-heading primitive.

## Release / migration impact

None. Pure refactor, no settings/data shape changes, `main.js`/`manifest`/`versions` unaffected by behavior.

---

<a href="https://app.devin.ai/review/chhoumann/quickadd/pull/1404" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **Refactor**
  * Improved reliability of capture destination parsing for “Capture to” inputs (vault, folders, files, properties, and filters), including clearer handling of ambiguous names and invalid targets.
  * Refined capture insertion positioning (insert-after/before, ordered creation, and inline anchors) for more consistent placement.
  * Improved rendering behavior for `{{FILE:...}}` tokens via a dedicated renderer.

* **Tests**
  * Added coverage for capture-target resolution, insertion positioning, and file-token rendering to validate edge cases and error handling.

### CODE REVIEW

File: src/formatters/helpers/insertionPositioning.ts [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+export function maskNonBodyHeadingsForSearch(
+	lines: string[],
+	fileContent: string,
+): string[] {
+	const bodyStartLine = getBodyStartLine(fileContent);
+	const masked = maskFencedHeadings(lines).map((line) =>
+		line.replace(/\r$/, ""),
+	);
+	for (let i = 0; i < bodyStartLine && i < masked.length; i++) {
+		masked[i] = "";
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/formatters/helpers/insertionPositioning.ts]: Good catch — valid pre-existing edge case (the masking ran over the frontmatter before it was blanked). Fixed in 81074563 by blanking frontmatter lines first and running fence masking on the body-scoped lines; byte-identical for fence-free frontmatter. Added a RED/GREEN regression test.
